include/irr/builtin/material_compiler/glsl/rasterization/impl.glsl

```diff
@@ -3,7 +3,7 @@
 
 #include <irr/builtin/material_compiler/glsl/common.glsl>
 
-void instr_eval_execute(in instr_t instr, in irr_glsl_LightSample s, in irr_glsl_AnisotropicMicrofacetCache uf)
+void instr_eval_execute(in instr_t instr, inout irr_glsl_LightSample s, inout irr_glsl_AnisotropicMicrofacetCache _uf, inout bool ts_flag)
```

Comment:
`_uf` should not be `inout`

the cache for a reflective ray is 100% correct always and wouldn't change

if you have a `L` which is behind the `N` and surface is not twosided (you actually get into a non-thin dielectric AST node) then you need to compute the microfacet cache for a transmissive case.

Reply:
it is `inout` only because of possible handling of twosided

Reply:
microfacet cache computed for transmission doesnt get out of this function

Reply:
ok I get it now.